### PR TITLE
HUE-9605 [core] change_default_password config Not working in latest hue versions

### PR DESCRIPTION
**Pull request by Akhilsnaik**
**What changes were proposed in this pull request?**
HUE-9605 [core] change_default_password config Not working in latest hue versions

**How was this patch tested?**

Tested the Scenarios 

- when change_default_password is true the change password loads well
-  Change the Password for an newly created user when change_default_password is true (no impact)
-  Change the Password for an newly created user when change_default_password is false ( no impact)
- Admin is able to change his password in profile page ( no impact)

The Fix Works Fine.

the page that is loaded for Mandatory Change password when change_default_password is set as true : 

Screenshot 2020-12-03 at 4 14 45 PM

### CODE REVIEW

File: apps/useradmin/src/useradmin/templates/change_password.mako [hunk 1]
```diff
@@ -64,72 +67,30 @@ ${ layout.menubar(section='users') }
...
-        $(this).parent().find(".help-inline").remove();
-        $(this).after("<span class=\"help-inline\"><strong>${ _('This field is required.') }</strong></span>");
+  var $editUserComponents = $('#editUserComponents');
+  % if is_embeddable:
+  $editUserComponents.find('#editForm').attr('action', window.location.pathname.substr((window.HUE_BASE_URL + '/hue').length).replace(/\/$/, ''));
+  $editUserComponents.find('#editForm').ajaxForm({
+    dataType:  'json',
+    success: function(data) {
+      if (data && data.status == -1) {
+        renderUseradminErrors(data.errors);
```
Comment (JohanAhlen) [inline apps/useradmin/src/useradmin/templates/change_password.mako:80]: You'll need to add this functions as well.

**Review by romainr — APPROVED** [review]
Just put some thoughts that might save time for later

File: desktop/core/src/desktop/auth/views.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def dt_login(request, from_modal=False):
           LOG.error('Could not create home directory at login for %s.' % user, exc_info=e)
 
         if require_change_password(userprofile):
-          return HttpResponseRedirect(urlresolvers.reverse('useradmin.views.edit_user', kwargs={'username': user.username}))
+          return HttpResponseRedirect('/hue' + urlresolvers.reverse('useradmin.views.edit_user', kwargs={'username': user.username}))
```
[the remark this answers is not in the document]
Comment (JohanAhlen) [reply desktop/core/src/desktop/auth/views.py:158]: I believe the reason is that we never converted this particular page when going from Hue 3 to 4, onePageViewModel will only pick up URLs after '/hue'.

File: apps/useradmin/src/useradmin/templates/change_password.mako (hunk 1 above)
Comment (Akhilsnaik) [reply apps/useradmin/src/useradmin/templates/change_password.mako:80]: Hi @JohanAhlen , the function `renderUseradminErrors` are already part layout.mako and we are importing that in our change_password.mako. dont think this is required.: 
https://github.com/cloudera/hue/blob/master/apps/useradmin/src/useradmin/templates/layout.mako#L107

File: apps/useradmin/src/useradmin/views.py [hunk 3]
```diff
@@ -400,7 +400,7 @@ def edit_user(request, username=None):
         if is_admin(request.user):
           return redirect(reverse('about:index'))
         else:
-          return redirect(reverse('desktop_views_home'))
+          return JsonResponse({'url': '/hue'})
```
[the remark this answers is not in the document]
Comment (Akhilsnaik) [reply apps/useradmin/src/useradmin/views.py:403]: tried that , the redirection was not happening and page reloads to the same edit page which confuses users and they might change password twice, so i am redirecting to /hue.

File: apps/useradmin/src/useradmin/templates/change_password.mako (hunk 1 above)
Comment (JohanAhlen) [reply apps/useradmin/src/useradmin/templates/change_password.mako:80]: Great, thanks!

**Review by JohanAhlen — APPROVED** [review]

**Pull request merged**